dali/operators/image/resize/tensor_resize_attr.cc

```diff
@@ -0,0 +1,251 @@
+// Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/operators/image/resize/tensor_resize_attr.h"
+#include <limits>
+#include <string>
+#include <vector>
+#include "dali/core/math_util.h"
+#include "dali/operators/image/resize/resize_attr.h"
+#include "dali/pipeline/operator/common.h"
+
+namespace dali {
+
+DALI_SCHEMA(TensorResizeAttr)
+    .AddOptionalArg<vector<float>>("sizes", R"code(Output sizes.
+
+When ``axes`` is provided, the size values refer to the axes specified.
+Note: Arguments ``sizes`` and ``scales`` are mutually exclusive.)code",
+                                   {}, true)
+    .AddOptionalArg<vector<float>>("scales", R"code(Scale factors.
+
+The resulting output size is calculated as
+``out_size = size_rounding(scale_factor * original_size)``.
+See ``size_rounding`` for a list of supported rounding policies.
+
+When ``axes`` is provided, the scale factor values refer to the axes specified.
+Note: Arguments ``sizes`` and ``scales`` are mutually exclusive.)code",
+                                   {}, true)
+    .AddOptionalArg<int>("alignment", R"code(Determines the position of the ROI 
+when using scales (provided or calculated).
+
+Accepted values are -1 (align with top-left corner), 0 (centered), 1 (align with bottom-right corner).
+By default, 0 (centered) is assumed. Contains as many elements as dimensions provided for sizes/scales.
+If only one value is provided, it is apply to all dimensions.)code",
+                                   std::vector<int>{0}, true)
+    .AddOptionalArg(
+        "axes",
+        R"code(Indices of dimensions that `sizes`, `scales`, `max_size`, `roi_start`, `roi_end` refer to.
+
+By default, all dimensions are assumed.)code",
+        std::vector<int>{})
+    .AddOptionalArg<std::string>("size_rounding",
+                                 R"code(Determines the rounding policy when using scales.
+
+Possible values are:
+* | ``"round"`` - Rounds the resulting size to the nearest integer value, with halfway cases rounded away from zero.
+* | ``"truncate"`` - Discards the fractional part of the resulting size.
+* | ``"ceil"`` - Rounds up the resulting size to the next integer value.)code",
+                                 "truncate")
```

Comment:
why not `"round"`? Don't we round in normal Resize?

Reply:
true, by default we round. I'll change it